File: __tests__/__fixtures__/output/node/fetch/full.js
const { URLSearchParams } = require('url');
const fetch = require('node-fetch');
const encodedParams = new URLSearchParams();

encodedParams.set('foo', 'bar');

const url = 'https://httpbin.org/anything?foo=bar&foo=baz&baz=abc&key=value';
const options = {
  method: 'POST',
  headers: {
    cookie: 'foo=bar; bar=baz',
    accept: 'application/json',
    'content-type': 'application/x-www-form-urlencoded'
  },
  body: encodedParams
};

fetch(url, options)
  .then(res => res.json())
  .then(json => console.log(json))
  .catch(err => console.error('error:' + err));

File: src/targets/java/asynchttp/fixtures/full.java
AsyncHttpClient client = new DefaultAsyncHttpClient();
client.prepare("POST", "https://httpbin.org/anything?foo=bar&foo=baz&baz=abc&key=value")
  .setHeader("cookie", "foo=bar; bar=baz")
  .setHeader("accept", "application/json")
  .setHeader("content-type", "application/x-www-form-urlencoded")
  .setBody("foo=bar")
  .execute()
  .toCompletableFuture()
  .thenAccept(System.out::println)
  .join();

client.close();